src/pocketmine/item/Food.php

```diff
@@ -41,7 +42,14 @@ public function getAdditionalEffects() : array{
 		return [];
 	}
 
-	public function onConsume(Living $consumer){
+	public function canBeConsumedBy(Living $consumer): bool{
+		return ($this->requiresHunger() and $consumer instanceof Human and !$consumer->isHungry()) ? false : true;
+	}
 
+	public function onConsume(Living $consumer){
+		if($consumer instanceof Human) {
+			$consumer->addFood($this->getFoodRestore());
+			$consumer->addSaturation($this->getSaturationRestore());
+		}
```

Comment:
The placement of this logic is now inconsistent with other consume handling. I also don't see why it's necessary to move this.